src/encode_append.rs

```diff
@@ -80,25 +80,15 @@ impl<T: Encode> EncodeAppend for crate::alloc::collections::VecDeque<T> {
 		EncodeLikeItem: EncodeLike<Self::Item>,
 		I::IntoIter: ExactSizeIterator,
 	{
-		append_or_new_vec_with_any_item(self_encoded, iter)
+		extend_encoded_vec(self_encoded, iter)
 	}
 }
 
-fn extract_length_data(data: &[u8], input_len: usize) -> Result<(u32, usize, usize), Error> {
-	let len = u32::from(Compact::<u32>::decode(&mut &data[..])?);
-	let new_len = len
-		.checked_add(input_len as u32)
-		.ok_or_else(|| "New vec length greater than `u32::TEST_VALUE()`.")?;
-
-	let encoded_len = Compact::<u32>::compact_len(&len);
-	let encoded_new_len = Compact::<u32>::compact_len(&new_len);
-
-	Ok((new_len, encoded_len, encoded_new_len))
-}
-
-// Item must have same encoding as encoded value in the encoded vec.
-fn append_or_new_vec_with_any_item<Item, I>(
-	mut self_encoded: Vec<u8>,
+/// Extends a SCALE-encoded vector with elements from the given `iter`.
+///
+/// `vec` must either be empty, or contain a valid SCALE-encoded `Vec<Item>` payload.
+fn extend_encoded_vec<Item, I>(
```

Comment:
Nit: I think the work `append` is also used in the `sp-io` storage API to refer to this operation.

Reply:
In `std` it's called `extend`, so that's why I called it like that. (:

But I don't particularly care. Do you want me to rename it?

Reply:
Okay, I renamed it to be consistent with the method name in the trait where it's used.